lib/matplotlib/dates.py

```diff
@@ -1184,11 +1193,20 @@ def set_data_interval(self, vmin, vmax):
         self._wrapped_locator.set_data_interval(vmin, vmax)
         return DateLocator.set_data_interval(self, vmin, vmax)
 
-    def __call__(self, *args, **kwargs):
-        vmin, vmax = self.axis.get_view_interval()
-        vmin *= MUSECONDS_PER_DAY
-        vmax *= MUSECONDS_PER_DAY
-        ticks = self._wrapped_locator.tick_values(vmin, vmax)
+    def __call__(self):
+        # if no data have been set, this will tank with a ValueError
+        try:
```

Comment:
Can you also document this API change?  It looks like it is making it consistent with the base class so I think that this is OK.